packages/framework/ai-collab/src/test/explicit-strategy/agentEditing.spec.ts

```diff
@@ -0,0 +1,271 @@
+// eslint-disable-next-line import/no-internal-modules
+import type { ResponseFormatJSONSchema } from "openai/resources/shared.mjs";
+
+// eslint-disable-next-line import/no-internal-modules
+import { objectIdKey } from "../../explicit-strategy/agentEditTypes.js";
+// eslint-disable-next-line import/no-internal-modules
+import { IdGenerator } from "../../explicit-strategy/idGenerator.js";
+// import { getResponse } from "../../explicit-strategy/llmClient.js";
+import {
+	getPromptFriendlyTreeSchema,
+	toDecoratedJson,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../explicit-strategy/promptGeneration.js";
+
+const demoSf = new SchemaFactory("agentSchema");
+
+class Vector extends demoSf.object("Vector", {
+	x: demoSf.number,
+	y: demoSf.number,
+	z: demoSf.optional(demoSf.number),
+}) {}
+
+class RootObject extends demoSf.object("RootObject", {
+	str: demoSf.string,
+	vectors: demoSf.array(Vector),
+	bools: demoSf.array(demoSf.boolean),
+}) {}
+
+const factory = SharedTree.getFactory();
+
+describe("toDecoratedJson", () => {
+	let idGenerator: IdGenerator;
+	beforeEach(() => {
+		idGenerator = new IdGenerator();
+	});
+	it("adds ID fields", () => {
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: Vector }));
+		view.initialize({ x: 1, y: 2 });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({
+				[objectIdKey]: "Vector1",
+				x: 1,
+				y: 2,
+			}),
+		);
+	});
+
+	it("handles nested objects", () => {
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: RootObject }));
+		view.initialize({ str: "hello", vectors: [{ x: 1, y: 2, z: 3 }], bools: [true] });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({
+				[objectIdKey]: "RootObject1",
+				str: "hello",
+				vectors: [
+					{
+						[objectIdKey]: "Vector1",
+						x: 1,
+						y: 2,
+						z: 3,
+					},
+				],
+				bools: [true],
+			}),
+		);
+
+		assert.equal(idGenerator.getNode("RootObject1"), view.root);
+		assert.equal(idGenerator.getNode("Vector1"), view.root.vectors.at(0));
+
+		// const hydratedObject = hydrate(
+		// 	RootObject,
+		// 	new RootObject({ str: "hello", vectors: [{ x: 1, y: 2, z: 3 }], bools: [true] }),
+		// );
+		// assert.equal(
+		// 	toDecoratedJson(idGenerator, hydratedObject),
+		// 	JSON.stringify({
+		// 		[objectIdKey]: "RootObject0",
+		// 		str: "hello",
+		// 		vectors: [
+		// 			{
+		// 				[objectIdKey]: "Vector0",
+		// 				x: 1,
+		// 				y: 2,
+		// 				z: 3,
+		// 			},
+		// 		],
+		// 		bools: [true],
+		// 	}),
+		// );
+		// assert.equal(idGenerator.getNode("RootObject0"), hydratedObject);
+		// assert.equal(idGenerator.getNode("Vector0"), hydratedObject.vectors.at(0));
```

Comment:
Remove? Tests below also have commented out bits.

Reply:
Removed in commit de57af318b60bf532498f8d7f9464d064454ef1e